kernel/src/scan/data_skipping.rs

```diff
@@ -23,6 +23,66 @@ use crate::{
 #[cfg(test)]
 mod tests;
 
+/// Transforms schema fields to be nullable, as stats may not be available for all columns
+/// (and usually aren't for partition columns).
+struct NullableStatsTransform;
+
+impl<'a> SchemaTransform<'a> for NullableStatsTransform {
+    fn transform_struct_field(&mut self, field: &'a StructField) -> Option<Cow<'a, StructField>> {
+        use Cow::*;
+        let field = match self.transform(&field.data_type)? {
+            Borrowed(_) if field.is_nullable() => Borrowed(field),
+            data_type => Owned(StructField {
```

Comment:
let's create a `with_nullable` function to change the nullability, similar to `StuctField::with_name`